public/API/API_GetUserFriendList.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ *  API_GetUserFriendList - returns a list of Friends, with basic data, for a user
+ *    u : username
```

Comment:
If I can only fetch my own friends list, why do I need to specify the `u` parameter?

Reply:
Ah, I see:
> With respect to viewing other users' lists, this works the same way that we set up API_GetWantToPlayList as well. A target user's friend list can be viewed if the calling user and target user are friends.

Although, I'm thinking this should probably be more restricted. I want to say this endpoint should only allow for fetching the caller's friends list.

Reply:
I agree. You should only be able to fetch your own friends list.

See also https://github.com/RetroAchievements/RAWeb/issues/127